### PR TITLE
bpo-34706:  Preserve subclassing in Signature.from_callable 

### PR DESCRIPTION
**Pull request by bukzor**
https://bugs.python.org/issue34706

Specifically in the case of a class that does not override its constructor signature inherited from object.

https://bugs.python.org/issue34706

### CODE REVIEW

**Comment by the-knights-who-say-ni** [comment]
Hello, and thanks for your contribution!

I'm a bot set up to make sure that the project can legally accept your contribution by verifying you have signed the [PSF contributor agreement](https://www.python.org/psf/contrib/contrib-form/) (CLA).

Unfortunately we couldn't find an account corresponding to your GitHub username on [bugs.python.org](https://bugs.python.org/) (b.p.o) to verify you have signed the CLA (this might be simply due to a missing "GitHub Name" entry in your b.p.o account settings). This is necessary for legal reasons before we can look at your contribution. Please follow [the steps outlined in the CPython devguide](https://devguide.python.org/pullrequest/#licensing) to rectify this issue.


You can [check yourself](https://check-python-cla.herokuapp.com/) to see if the CLA has been received.

Thanks again for your contribution, we look forward to reviewing it!

**Review by 1st1 — APPROVED** [review]

**Comment by 1st1** [comment]
Please add a NEWS entry and I'll merge this. Thanks!

**Review by asottile — APPROVED** [review]

File: Lib/test/test_inspect.py [hunk 1]
```diff
@@ -3089,6 +3089,13 @@ def foo(a, *, b:1): pass
         foo_sig = MySignature.from_callable(foo)
         self.assertTrue(isinstance(foo_sig, MySignature))
 
+    def test_signature_from_callable_class(self):
+        # A regression test for a class inheriting its signature from `object`.
+        class MySignature(inspect.Signature): pass
+        class foo: pass
+        foo_sig = MySignature.from_callable(foo)
+        self.assertTrue(isinstance(foo_sig, MySignature))
```
Comment (serhiy-storchaka) [inline Lib/test/test_inspect.py]: Use `assertIsinstance()`.

**Review by serhiy-storchaka — COMMENTED** [review]
Is this a bug fix or adding a new feature? In the former case it should be backported to older Python versions. In the latter case it needs a `versionchanged` directive in the documentation.

**Comment by bukzor** [comment]
> Please add a NEWS entry and I'll merge this. Thanks!

I did see that notice, but couldn't figure out what to write, or whether this change is notable.
I was hoping that a maintainer would add it to my PR, since I've allowed edits.

What should the NEWS entry say? "Preserve subclassing in Signature.from_callable" ?

File: Lib/test/test_inspect.py (hunk 1 above)
Comment (bukzor) [reply Lib/test/test_inspect.py]: This is the style used in this file. AssertIsinstance is not used at all. Shall I change the rest of the file or only this line? Such cleanup may be better as a separate PR.

Comment (gpshead) [reply Lib/test/test_inspect.py]: we use the more modern assertion methods when writing new code.  don't worry about the rest of the file.

**Comment by the-knights-who-say-ni** [comment]
Hello, and thanks for your contribution!

I'm a bot set up to make sure that the project can legally accept this contribution by verifying everyone involved has signed the [PSF contributor agreement](https://www.python.org/psf/contrib/contrib-form/) (CLA).



## CLA Missing

Our records indicate the following people have not signed the CLA:

@mrkn

For legal reasons we need all the people listed to sign the CLA before we can look at your contribution. Please follow [the steps outlined in the CPython devguide](https://devguide.python.org/pullrequest/#licensing) to rectify this issue.

If you have recently signed the CLA, please wait at least one business day
before our records are updated.


You can [check yourself](https://check-python-cla.herokuapp.com/) to see if the CLA has been received.

Thanks again for the contribution, we look forward to reviewing it!

**Comment by gpshead** [comment]
my push failed to do the right thing, i'll re-create this PR.

**Pull request closed without merging**